test/EventLoopTest.php

```diff
@@ -298,6 +298,50 @@ public function testSuspensionThrowingErrorViaInterrupt(): void
         } catch (UncaughtThrowable $t) {
             self::assertSame($error, $t->getPrevious());
         }
+
+        $suspension->resume(); // Calling resume on the same suspension should not throw an Error.
+        $suspension->throw(new \RuntimeException()); // Calling throw on the same suspension should not throw an Error.
+
+        try {
+            $suspension->suspend(); // Calling suspend on the same suspension should throw an Error.
+            self::fail("Error was not thrown");
+        } catch (\Error $e) {
+            self::assertStringContainsString('suspended after an uncaught exception', $e->getMessage());
+        }
+
+        // Creating a new Suspension and re-entering the event loop (e.g. in a shutdown function) should work.
+        $suspension = EventLoop::getSuspension();
+        EventLoop::queue($suspension->resume(...));
+        $suspension->suspend();
+    }
+
+    public function testSuspensionThrowingErrorViaInterrupt2(): void
+    {
+        $suspension = EventLoop::getSuspension();
+        $error = new \Error("Test error");
+        EventLoop::queue(static fn () => throw $error);
+        EventLoop::queue($suspension->resume(...), 123);
+        try {
+            $suspension->suspend();
+            self::fail("Error was not thrown");
+        } catch (UncaughtThrowable $t) {
+            self::assertSame($error, $t->getPrevious());
+        }
+
+        $suspension->resume(); // Calling resume on the same suspension should not throw an Error.
```

Comment:
This should throw, because main isn't dead yet?

Reply:
I'm not sure this should throw if the {main} suspension which has been replaced. This may just cause uncaught loop exceptions during shutdown once again if the suspension is resumed by chance.

Reply:
This should run before the two queued callbacks above it and the suspension isn't replaced in this test, yet, or am I missing something?

Reply:
The call to `$suspension->suspend()` will switch from {main} to the event loop. {main} is then resumed by the uncaught exception from the callback passed to the first call to `EventLoop::queue()`. So at this call to `$suspension->resume()`, `$suspension->deadMain` will be true.

Reply:
Oh... I thought the entire try block was inside the queued callback. Not sure whether it's just because I'm reading this on my phone, but a newline between the queue calls and the try block would probably have helped. 😄